modules/apps/commerce/commerce-product-service/src/main/java/com/liferay/commerce/product/service/impl/CPOptionLocalServiceImpl.java

```diff
@@ -149,6 +155,32 @@ public CPOption deleteCPOption(CPOption cpOption) throws PortalException {
 		_cpOptionValueLocalService.deleteCPOptionValues(
 			cpOption.getCPOptionId());
 
+		// Commerce product option rels
+
+		CPDefinitionOptionRelLocalService cpDefinitionOptionRelLocalService =
+			_cpDefinitionOptionRelLocalServiceSnapshot.get();
+
+		long cpOptionId = cpOption.getCPOptionId();
+
+		int cpOptionCPDefinitionOptionRelsCount =
+			cpDefinitionOptionRelLocalService.
+				getCPOptionCPDefinitionOptionRelsCount(cpOptionId);
+
+		if (cpOptionCPDefinitionOptionRelsCount > 0) {
+			throw new CPDefinitionOptionRelException();
+		}
+
+		List<CPDefinitionOptionRel> cpDefinitionOptionRels =
+			cpDefinitionOptionRelLocalService.getCPOptionCPDefinitionOptionRels(
+				cpOptionId);
+
+		for (CPDefinitionOptionRel cpDefinitionOptionRel :
+				cpDefinitionOptionRels) {
+
```

Comment:
this will never get run, right? I think you can delete these lines of code

Reply:
You're right, thank you!